srtcore/channel.cpp

```diff
@@ -102,7 +102,8 @@ m_iIpTTL(-1),   /* IPv4 TTL or IPv6 HOPs [1..255] (-1:undefined) */
 m_iIpToS(-1),   /* IPv4 Type of Service or IPv6 Traffic Class [0x00..0xff] (-1:undefined) */
 #endif
 m_iSndBufSize(65536),
-m_iRcvBufSize(65536)
+m_iRcvBufSize(65536),
+m_bIpV6Only(1)
```

Comment:
Setting IPv6Only default to true breaks existing applications making use of IPv4mapped addresses. Also if SRT sockets  behavior is the inverse of natural sockets this is confusing.

Reply:
Ok you're right. In fact this correction was here to allow setting IPV6_V6ONLY to true though the default value depends on the system :
 - [Windows set it by default to true](https://docs.microsoft.com/en-us/windows/desktop/winsock/ipproto-ipv6-socket-options)
 - [Linux set it by default to false](http://man7.org/linux/man-pages/man7/ipv6.7.html)

Reply:
It is not that simple, the Linux system default is configurable (/proc/sys/net/ipv6/bindv6only) so either way in SRT (true or false) some application will break upon upgrade. To preserve system default which may be the result of admin policy, we should have no default (or match running system default) and only set channel if the program specifically set SRTO_IPV6ONLY.

Reply:
I have just done a fix for this purpose.

Reply:
The fix seems ok to me, for seamless upgrade.

Reply:
I have one doubt, in fact the assignment is always done there : 
https://github.com/Haivision/srt/blob/7f1bbf931be58ae4220cc4c4945683d8d3140cff/srtcore/api.cpp#L1693
I guess that there should be another condition here to check if the option has really been set. And we should change to an integer too. What do you think?

Reply:
You are right, I only looked at the last commit. And this raise another concern. The SRTO option is set in the SRT socket that sets it in the possibly shared channel (Mux). Therefore the IPV6_V6ONLY option must be added to the tests definng if an existing channel can be shared, like IPversion, MSS, IPTTL and IpToS.

Reply:
So it means that it must be added to the CMultiplexer too? 
And I am not sure if it should be reseted in CUDT::clearData(), what's your opinion?

Reply:
Not necessarily, Have a look at the code in api.cpp lines 1650 to 1658 where sockoptions are compared.